src/nixos.rs

```diff
@@ -66,6 +68,28 @@ enum OsRebuildVariant {
   Boot,
   Test,
   BuildVm,
+  BuildImage,
+}
+
+impl OsBuildImageArgs {
+  fn build_image(self, elevation: ElevationStrategy) -> Result<()> {
```

Comment:
The function body directly formats user input into an attribute path without any validation. This will produce cryptic errors if the variant doesn't exist. Add validation or at least document what valid variants are. Mind you, most variants are not available until the user *actually* adds a profile or exposes them, so this command will fail more often than not.